File: day-4/part-1.js
const { read } = require('../utils')

;(async () => {
  const data = await read('./data.txt', '\n\n')

  const requiredFields = [
    'byr',
    'iyr',
    'eyr',
    'hgt',
    'hcl',
    'ecl',
    'pid'
  ]

  let numValidPassports = 0
  data.forEach(datum => {
    const formatted = datum.replace(/\n/g, ' ')
    const fields = formatted.split(' ')

    let fieldCount = 0
    fields.forEach(field => {
      console.log('field', field)
      const [ key ] = field.split(':')
      if (requiredFields.includes(key)) {
        fieldCount += 1
      }
    })

    if (fieldCount === requiredFields.length) {
      numValidPassports += 1
    }
  })
  
  console.log('num valid passports: ', numValidPassports)
})()

File: day-4/part-1.py
with open('data.txt') as f:
  data = f.read().split('\n\n')

  requiredFields = [
    'byr',
    'iyr',
    'eyr',
    'hgt',
    'hcl',
    'ecl',
    'pid'
  ]

  numValidPassports = 0
  
  for datum in data:
    formatted = datum.replace('\n', ' ')
    fields = formatted.split(' ')

    fieldCount = 0
    for field in fields:
      key, *_ = field.split(':')
      if key in requiredFields:
        fieldCount += 1

    if fieldCount == len(requiredFields):
      numValidPassports += 1
  
  print(numValidPassports)